pkg/agent/controller/egress/ip_scheduler.go

```diff
@@ -200,6 +273,47 @@ func (o EgressesByCreationTimestamp) Less(i, j int) bool {
 	return o[i].CreationTimestamp.Before(&o[j].CreationTimestamp)
 }
 
+// updateMaxEgressIPsByNode updates the maxEgressIPs for a given Node in the cache.
+// It returns whether there is a real change, which indicates if rescheduling is required.
+func (s *egressIPScheduler) updateMaxEgressIPsByNode(nodeName string, maxEgressIPs int) bool {
+	s.nodeToMaxEgressIPsMutex.Lock()
+	defer s.nodeToMaxEgressIPsMutex.Unlock()
+
+	oldMaxEgressIPs, exists := s.nodeToMaxEgressIPs[nodeName]
```

Comment:
If not exists, should we return false if the value equals to the global value?

Reply:
Yes, and skipped inserting it to avoid extra check when deleting the cached value (as there is no need to reschedule when deleting the same value).